### ISSUE TITLE
Being able to turn and move based on different parameters.

### ISSUE THREAD

**Issue by paulmorio** [ChrstSea/sdp#11]
TLDR:
[MILESTONE2] For the planner I need (will be edited based on whether we have it, or I add more).
- Ability to turn for a set amount of time, or degree (probably time because of time constraints)
- Ability to go forwards/backwards for a set amount of time or distance.

Holonomic implementations will have to come a bit later(?)

In detail:
For all plans being made, it is necessary to find the angle in which to turn to and displacement of the ball for each of the robots. This can be done easily by calling the relevant pitch objects methods through the world being used by the planner.

However the problem is when giving commands to the robot in that plan cycle, as curently it is only possible to give commands such as `TURN-RIGHT` which will make the robot turn towards the right indefinitely until the robot receives a new plan (which might make it turn left, and then the cycle will keep on going). This is the implementation used currently (excerpt below)

``` Python
    elif (self.mode == 'dog'):

        # If the robot does not have the ball, it should go to the ball.
        if (self.state == 'noBall'):
            # Get the ball position so that we may find the angle to align with it, as well as the displacement
            ball_x = self.world._ball.x()
            ball_y = self.world._ball.y()

            angle_to_turn_to = self.bot.get_rotation_to_point(ball_x,ball_y)
            distance_to_move = self.bot.get_displacement_to_point(ball_x, ball_y)

            dir_to_turn = get_rotation_direction(self.world._ball)

            # We command the robot turn to the ball, and move forward if it is facing it.
            # This is implementation is deeply simplified (but works)
            bot_rotate_or_move(dir_to_turn)
```

As information can be calculated in the planner (like how much to turn, how much to go forward, etc.) it would be advantageous to be able to pass commands that utilise this information. So something like: 

``` Python
    elif (self.mode == 'dog'):

        # If the robot does not have the ball, it should go to the ball.
        if (self.state == 'noBall'):
            # Get the ball position so that we may find the angle to align with it, as well as the displacement
            ball_x = self.world._ball.x()
            ball_y = self.world._ball.y()

            angle_to_turn_to = self.bot.get_rotation_to_point(ball_x,ball_y)
            distance_to_move = self.bot.get_displacement_to_point(ball_x, ball_y)

            dir_to_turn = get_rotation_direction(self.world._ball)

            # We command the robot turn to the ball, and move forward if it is facing it.
            bot_rotate_or_move(dir_to_turn, angle_to_turn_to, distance_to_move)
```
